W-13804500 error handlers cv (#92)
mulesoft/docs-code-builder: modules/ROOT/pages/_partials/mule-components/4.4/component-error-handler.adoc

//
//tag::component-title[]

= Error Handler (<error-handler/>)

//end::component-title[]
//

//
//tag::component-short-description[]
//     Short description of the form "Do something..." 
//     Example: "Configure log messages anywhere in a flow."

Catches an error thrown by the Mule application and routes it to the first On-Error component configuration that matches the error type. 

//end::component-short-description[]
//

//
//tag::component-long-description[]

The Error Handler component has two child elements, On Error Continue and On Error Propagate, which enable you to configure routes that execute when the error raised in the application matches the type of error configured in the handler. Each handler has its own behavior and works relative to the component that contains it (a Flow or a Try scope):

* On Error Continue (`on-error-continue`): executes and uses the result of the execution as the result of its container element (as if the container completed the execution successfully). Any transactions at this point commit successfully.

> * On Error Propagate (`on-error-propagate`): rolls back any transactions, executes, and uses that result to re-throw the existing error. In this case, the container element "fails" its execution.

Reviewer: ```suggestion
* On Error Propagate (`on-error-propagate`): rolls back any transactions, executes, and uses that result to rethrow the existing error. In this case, the container element "fails" its execution.
```

      </on-error-continue>
    </error-handler>

</flow>
----

//end::component-xml[]
//
//tag::component-snippet-xml[]

[[snippet]]

Snippet XML in Anypoint Code Builder serves as a customizable template for the component. 

[source,xml]
----
SNIPPET_XML_HERE
----

//end::component-snippet-xml[]
//
//
//
//
//TABLE: ROOT XML ATTRIBUTES (for the top-level (root) element)
//tag::component-xml-attributes-root[]

[%header,cols="1,3a"]
|===
| `<error-handler/>` Attributes 
| Description

| `doc:name` 
| Editable name for the component.

| `doc:id` 
| Automatically generated identifier for the component.

|===
//end::component-xml-attributes-root[]
//
//
//TABLE (IF NEEDED): CHILD XML ATTRIBUTES for each child element
//  Repeat as needed, adding the next number to the tag value. 
//  Provide intro text, as needed.
//tag::component-xml-attributes-child1[]

=== Child XML Elements and Attributes

The `<on-error-continue/>` and `<on-error-propagate/>` child elements share the following attributes:

[%header, cols="1,3"]
|===
|Attributes | Description

| `type` | A comma-separated list of Mule error types that defines the errors this on-error strategy handles.

> | `when` | A matching expression that defines the condition to trigger the execution of this on-error strategy. The expression must evaluate to a boolean value (`true` or `false`).

Reviewer: ```suggestion
| `when` | A matching expression that defines the condition to trigger the execution of this on-error strategy. The expression must evaluate to a Boolean value (`true` or `false`).
```

> | `enableNotifications` | Specifies if this strategy fires a notification of type `ExceptionNotification` when an error occurs. Defaults to `true`.

Reviewer: ```suggestion
| `enableNotifications` | Specifies if this strategy triggers a notification of type `ExceptionNotification` when an error occurs. Defaults to `true`.
```
